rustls/examples/internal/bogo_shim_impl.rs

```diff
@@ -1149,6 +1155,46 @@ fn exec(opts: &Options, mut sess: Connection, count: usize) {
             );
         }
 
+        if let Some(curve_id) = &opts.expect_curve_id {
+            // unlike openssl/boringssl's API, `negotiated_key_exchange_group`
+            // works for the connection, not session.  this means TLS1.2
+            // resumptions never have a value for `negotiated_key_exchange_group`
+            let negotiated_key_exchange_group_ready = !(sess.is_handshaking()
+                || sess.protocol_version().unwrap() == ProtocolVersion::TLSv1_2
+                    && sess.handshake_kind().unwrap() == HandshakeKind::Resumed);
```

Comment:
nit: I think this would be easier to parse if it were broken up a little bit:
```rust
let tls12_resumed = sess.protocol_version().unwrap() == ProtocolVersion::TLSv1_2 && sess.handshake_kind().unwrap() == HandshakeKind::Resumed;
let negotiated_key_exchange_group_ready = !(sess.is_handshaking() || tls12_resumed)
```
```